### DISCUSSION TITLE
How to update version in Pyproject.toml?

### DISCUSSION THREAD

**Discussion by cemrehancavdar** [sanders41/cookiecutter-python#300] [General]
Should we update version number in pyproject.toml by hand before releasing a new verstion to pypi? 
Are there any way to automaticly increase it with release version? How do you do it?

**Comment by sanders41** [comment]
I do it by hand before releasing.

I have working on a project that used [versioneer](https://github.com/python-versioneer/python-versioneer) automate this so it is possible. Personally I prefer by hand over versioneer because versioneer used the git history to generate the version and it would cause unexpected issues at times.

**Reply by cemrehancavdar** [reply]
Understood, I wonder is there a way to change version number via release name without 'v' via poetry version $version_number or smth in workflow

**Reply by sanders41** [reply]
I've looked around for something like this but haven't found anything. To be honest though I haven't looked very hard so it's possible there is something I just haven't found.

**Comment by cemrehancavdar** [comment]
I have managed to do it by adding following codes. It uses tag_name as version. It adds version with "v" but it pushes correcty to pypi. 

    - name: Install Dependencies
      run: |
        poetry install
------------------------------
    - name: Update Version
      run: |
        poetry version ${{ github.event.release.tag_name }}
    - name: setup git config
      run: |
          # setup the username and email. I tend to use 'GitHub Actions Bot' with no email by default
          git config user.name "GitHub Actions Bot"
          git config user.email "<>"

    - name: commit
      run: |
          # Stage the file, commit and push
          git add pyproject.toml
          git commit -m "update version: ${{ github.event.release.tag_name }}"
          git push origin master

**Reply by sanders41** [reply]
You could possibly remove the `v`. It works in the terminal at least, but I haven't tested it in GH actions.

In terminal

```sh
export version=v1.0.0
echo ${version:1}
```

So in GH actions maybe?

```yaml
poetry version ${{ github.event.release.tag_name:1 }}
```

**Comment by cemrehancavdar** [comment]
Thanks, it became breeze to deploy.
